### PR TITLE
feat: added fieldTypes prop to form

### PR DESCRIPTION
**Pull request by ReptoxX**
Introduce a way to configure the input types through a Form prop.

As mentioned in #217.

### CODE REVIEW

**Comment by ReptoxX** [comment]
@danielweinmann could you please review this pr? I want it to get merged if everything is correct.

**Review by danielweinmann — CHANGES_REQUESTED** [review]
Hey, @ReptoxX! Thanks for the PR and sorry for the delay in reviewing it. It's looking real good! I just requested a few minor changes and we'll be good to go!

File: apps/web/app/routes/examples/forms/edit-types.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+const code = `const schema = z.object({
+  email: z.string().min(1).email(),
+  password: z.string().min(8),
+})
+
+export default () => (
+  <Form
+    schema={schema}
+    fieldTypes={{
```
[the remark this answers is not in the document]
Comment (ReptoxX) [reply apps/web/app/routes/examples/forms/edit-types.tsx]: I hope my current example is enough.

**Comment by ReptoxX** [comment]
Hey @danielweinmann,
i included every wanted change. 

Hope this works out.

**Review by danielweinmann — CHANGES_REQUESTED** [review]
LGTM! Can you resolve the conflicts before we merge?

**Comment by ReptoxX** [comment]
Okay i've resolved the conflict.

**Review by danielweinmann — CHANGES_REQUESTED** [review]
@ReptoxX, can you fix the broken tests? It looks like it's just a label that's incorrect.

File: apps/web/tests/setup/example.ts [hunk 2]
```diff
@@ -25,13 +25,15 @@ class Example {
...
   readonly password: Field
+  readonly favColor: Field
   readonly button: Locator
 
   constructor(page: Page) {
     this.page = page
     this.firstName = this.field('firstName')
     this.email = this.field('email')
     this.password = this.field('password')
+    this.favColor = this.field('favColor')
```
Comment (danielweinmann) [inline apps/web/tests/setup/example.ts:36]: I had missed this one: please do not add this field to the default test example and add it only to your test file.

**Review by danielweinmann — CHANGES_REQUESTED** [review]

File: apps/web/tests/setup/example.ts (hunk 2 above)
Comment (ReptoxX) [reply apps/web/tests/setup/example.ts]: I'm sorry, yes i should've not done that.
It's fixed.

**Review by danielweinmann — APPROVED** [review]

**Comment by danielweinmann** [comment]
Thank you, @ReptoxX! 🎉

**Pull request merged**